### PR TITLE
COM-372: implement inline alert component

### PR DESCRIPTION
**Pull request by TheEnmanuel23**
## Please delete checklist items and sections that are not relevant

**Checklist**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Please check if the PR fulfills these requirements**

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Storybook or docs have been added / updated (for bug fixes / features)

**What is the ticket / issue associated with this change?** (Link to the ticket or issue)
https://impulsumvc.atlassian.net/browse/COM-372

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

**Other information**:

**Storybook Link** (Link to a local storybook url to review new functionality)

### CODE REVIEW

**Review by TheEnmanuel23 — COMMENTED** [review]
I reused Alert props and status array since they are basically  the same

File: packages/components/src/theme/ComposerTheme/Components/InlineAlert.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    return {
+      container: {
+        alignItems: 'center',
+        gap: '0.5rem',
+      },
+      message: {
+        ...messageStylesByStatus[status],
+      },
+      icon: {
+        ...rest.theme.components.Alert.baseStyle(rest).icon.status[status],
```
Comment (TheEnmanuel23) [inline packages/components/src/theme/ComposerTheme/Components/InlineAlert.ts:34]: I wanted to use somethig like this. as an object:
`...Alert.baseStyle.icon.status[status]`
I din't get anything, due to Alert theme is defined as function in its `baseStyles` property, so I changed it `...Alert.baseStyle(props).icon.status[status]`
but I was getting another error from typescript.

File: packages/components/src/components/notifications/InlineAlert/InlineAlert.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    status,
+  }) as InlineAlertStyles;
+
+  const BaseIcon = ALERT_ICONS[status];
+
+  return (
+    <Flex ref={ref} {...styles.container} {...rest}>
+      {showIcon && <BaseIcon {...styles.icon} data-testid="cmpsr.inline-alert.icon" />}
+      <Text {...styles.message}>{message}</Text>
+    </Flex>
```
Comment (tkanzakic) [inline packages/components/src/components/notifications/InlineAlert/InlineAlert.tsx:18]: this should use composition

**Review by tkanzakic — CHANGES_REQUESTED** [review]

File: packages/components/src/components/notifications/InlineAlert/InlineAlert.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import React, { FC, PropsWithChildren } from 'react';
+import { Flex, Text, ALERT_ICONS, InlineAlertContextProps } from '@components';
+import { forwardRef, useMultiStyleConfig } from '@chakra-ui/react';
+import { createContext } from '@chakra-ui/react-utils';
```
[the remark this answers is not in the document]
Comment (tkanzakic) [reply packages/components/src/components/notifications/InlineAlert/InlineAlert.tsx:4]: yes, it returns the provider and a hook to consume the context in the same call

**Review by guilean — APPROVED** [review]

**Review by tkanzakic — APPROVED** [review]

**Pull request merged**